pkg/plugin/server.go

```diff
@@ -43,7 +49,20 @@ func (s *KeyManagementServiceServer) Version(ctx context.Context, request *k8spb
 	}, nil
 }
 
+// Encrypt message
 func (s *KeyManagementServiceServer) Encrypt(ctx context.Context, request *k8spb.EncryptRequest) (*k8spb.EncryptResponse, error) {
+	start := time.Now()
+
+	var err error
+	defer func() {
+		if err != nil {
+			s.reporter.ReportEncryptErrorCountMetric(ctx, internalerrors.FailedToEncrypt)
```

Comment:
In the csi driver, we use the error type to signify a more specific error (provider error, grpc error). I think `FailedToEncrypt` is already know from the metric. Instead we should set the `error_type` to `AuthenticationFailed`, `NotFound` error. This will be useful to generate alerts.

Reply:
fixed. passing raw errors to metrics for better usage

Reply:
The errors being returned here are either custom errors(encoding etc) and errors from key vault client/SDK. I don't believe it will reveal any sensitive data. However they will be verbose. Not sure if we want to reveal that much details in metrics.
As far the metrics itself, we are reporting latency distribution and total count. These should not be sensitive. WDYT?

Reply:
We will monitor what information is being surfaced and then take actions accordingly in future releases.